tests/generation/models/copynet_test.py

```diff
@@ -330,3 +338,9 @@ def setup_method(self):
 
     def test_model_can_train_save_load_predict(self):
         self.ensure_model_can_train_save_and_load(self.param_file, tolerance=1e-2)
+
+    def test_multi_layer_decoder_model_can_train_save_and_load(self):
+        param_overrides = json.dumps({"model": {"target_decoder_layers": 3}})
+        self.ensure_model_can_train_save_and_load(
+            self.param_file, tolerance=1e-2, overrides=param_overrides
+        )
```

Comment:
I think we could do without this test. Not that it's a bad test, I just think we have everything covered already, and these tests have a lot of overhead.